compiler/damlc/daml-compiler/src/DA/Daml/Compiler/DataDependencies.hs

```diff
@@ -383,7 +384,7 @@ generateSrcFromLf env = noLoc mod
     -- | Generate instance declarations from dictionary functions.
     instanceDecls :: [Gen (Maybe (LHsDecl GhcPs))]
     instanceDecls = do
-        dval@LF.DefValue {..} <- NM.toList $ LF.moduleValues $ envMod env
+        dval@LF.DefValue {..} <- reverse . sortOn nameKey $ NM.toList $ LF.moduleValues $ envMod env
```

Comment:
```suggestion
        dval@LF.DefValue {..} <- sortOn (Down . nameKey) $ NM.toList $ LF.moduleValues $ envMod env
```

Reply:
It's kind of surprising that you have to sort them in descending order. Strange...